constraints/strongStart.js

```diff
@@ -36,6 +68,62 @@ function strongStart_Elfner(s, ptree, k){
 	return vcount;
 }
 
+/* Hsu 2016, p. 195
+	"STRONGSTART(k/p)
+	Assign a violation mark for every prosodic constituent whose leftmost daughter
+	constituent is of type k and is lower in the Prosodic Hierarchy than its sister
+	constituent immediately to the right, where k is at the left edge of a prosodic
+	constituent p.
+
+	The relevant notion of 'left edge' is defined as follows:
+	(57) A prosodic constituent k is at the left edge of prosodic constituent p iff.
+	a. p dominates k, and
+	b. no prosodic constituent that both dominates k and is dominated by p has a
+	leftmost daughter constituent that does not contain k."
+
+	[Feel free to create a paraphrase as well as keeping this quote!]
+	Note that the violations are for each parent p with k at its edge, not for every k.
+*/
+
+function strongStart_Hsu(s, ptree, k, p, node){
+
+	//since we cannot search up the tree, the original tree must be retained to determine whether a node of cat p dominates a node of cat k.
+	node = node || ptree;
+
+	//base case: node is a leaf or only has one child
+	if(!node.children){
+		return 0;
+	}
+	
+	var vcount = 0;
+	
+	// if node.children[0].cat === k and has a sibling, then compare it with its sibling as well as for domination by a node of cat p along the left edge.
+	if(node.children.length>1 && node.children[0].cat === k){		
+		if((pCat.isLower(node.children[0].cat, node.children[1].cat)) && seekCatLeftEdge(ptree, p, node.id)){ // searches tree for node of cat p dominating this node of cat k
```

Comment:
Shouldn't this be calling catDomsIdAtLeftEdge instead?

Reply:
Yes, you're right